apluslms-yamlidator/apluslms_yamlidator/document.py

```diff
@@ -325,10 +381,46 @@ def mlset(self, keys, value):
         container, key = find_ml(self._data, keys)
         container[key] = value
 
+    def mlset_cast(self, keys, value):
+        val_type = self.find_type(keys)
+
+        if val_type:
+            if val_type == "boolean":
+                try:
+                    value = to_bool(value)
+                except ValueError as err:
+                    err.value_type = val_type
+                    raise err
+            elif val_type == "integer" or val_type == "string":
+                try:
+                    value = int(value)
+                    # if conversion is successfull, the value is not a string
```

Comment:
I'm not sure if this is a good idea. There might be cases where one wants to add integer to a string field. For example, when adding environment variables. I would presume that this is less than 1% of the case, but still.

So, which kind of cases this check should solve or what is the goal for it?